### ISSUE TITLE
More training details

### ISSUE THREAD

**Issue by ChicyChen** [VTool-R1/VTool-R1#2]
Dear authors,

Could you please provide more training details? Thanks!

Best,
Chicy

**Comment by JingchengYang4** [comment]
We believe the information in the readme file is sufficient to conduct proper training, it may not be exhaustive, so if you have specific questions about specific issues and details, we are happy to answer it.

**Comment by ChicyChen** [comment]
Thanks for the response.
Could you provide the actual multinode training scripts? train_7b_table.sh should not directly work since verl requires using ray for multinode

**Comment by ChicyChen** [comment]
Besides, it seems training the 3B models with 8 A100 GPUs will take about 11-12 days, is that correct? Thank you!

**Comment by ChicyChen** [comment]
> Besides, it seems training the 3B models with 8 A100 GPUs will take about 11-12 days, is that correct? Thank you!

For this question, I estimate it to be 11-12 days since the default max step in tooluse_config_llm.yaml is 2000. However, I could not find experiment setting details on the total training steps/epochs in the paper, and I see Figure 3 shows up to about 50 steps only. Could you please clarify that? Thank you!

**Comment by JingchengYang4** [comment]
Hi, regarding the first point, I think the script would work directly it's just that it doesn't support multi-node out of the box. In a non-slurm environment, it's easier to first create a ray network with all your nodes attached, then run this script (on the head node) which will recognize the pre-existing ray network and use it for training.

You can refer to VERL's official doc on starting a multi-node ray cluster.
https://verl.readthedocs.io/en/latest/start/multinode.html

If you are using a slurm system, we do have a slurm script for that, note that you will need to make minor changes to accommodate for your specific slurm environment. I have uploaded an example slurm training script with 2x4 GPU config to the repo. You only need to change the SBATCH parameters for a different multi-node setup.

For training time, you do not need to train to the max step of 2000. The number is only meant to not pre-maturely interrupt our training. As you have noticed in our paper, it takes many less steps for the training to saturate. Usually for 3B, that is within 100 steps. Note however that steps are relative to the global batch size.  assuming you are using 80G 8xA100s (and not 40G models), which has a similar VRAM to the 8xH100 GPUs we used for 3B training, most of our training completes within 20-50 steps (assuming you do not make changes to the batch size). This takes about less than a day on 8xH100 systems but may very well take double the time on 8xA100, but should complete within a day or two regardless.

Please let us know if you have additional questions.

**Comment by Mingyuan1997** [comment]
Hi Chicy,

Thank you for your interests in our repo. We attach a plot for validation accuracy w.r.t training steps for 7B model in the readme for rough estimation of the training time. But the exact training time depends on your cluster, gpu and some configs such as batch size. Overall, we believe that the training can be done with academic-level resources.

Good luck on your training setup.

Best,

Mingyuan
